styx-common/src/main/java/com/spotify/styx/util/CachedSupplier.java

```diff
@@ -18,10 +18,9 @@
  * -/-/-
  */
 
-package com.spotify.styx;
+package com.spotify.styx.util;
```

Comment:
Would it make sense to use the Guava memoizing supplier instead of rolling our own?

http://google.github.io/guava/releases/21.0/api/docs/com/google/common/base/Suppliers.html#memoizeWithExpiration-com.google.common.base.Supplier-long-java.util.concurrent.TimeUnit-